clang/lib/CConv/ConstraintVariables.cpp

```diff
@@ -375,7 +375,8 @@ PointerVariableConstraint::PointerVariableConstraint(const QualType &QT,
         // inside a macro. Not sure how to handle this, so fall back to tyToStr.
         if (BaseType.empty())
           FoundMatchingType = false;
-      }
+      } else
+        FoundMatchingType = false;
```

Comment:
Seems odd to me that `FoundMatchingType` is the default, rather than the exception. There are multiple places you set it as `false`; would it make sense to default to `false` and set it to `true` when sure?